src/psyclone/psyGen.py

```diff
@@ -2803,12 +2803,27 @@ def gen_code(self, parent):
 
 
 class Kern(Call):
+    '''
+    Class representing a Kernel call within the Schedule (AST) of an Invoke.
+
+    :param type KernelArguments: the sub-class of KernelArguments to create.
```

Comment:
Is this not a more specific type of class?

Reply:
No, it is a `type`, not an object. However, I've extended the description to make it clear that the type will be a sub-class of `psyGen.Arguments`.

Reply:
You would have thought I would know that, seeing as I wrote the code - doh! Anyway the improved description is helpful.